tokio/src/sync/semaphore.rs

```diff
@@ -96,21 +88,21 @@ impl Semaphore {
     }
 
     /// Acquires permit from the semaphore.
-    pub async fn acquire(&self) -> SemaphorePermit<'_> {
-        self.ll_sem.acquire(1).await.unwrap();
-        SemaphorePermit {
+    pub async fn acquire(&self) -> Result<SemaphorePermit<'_>, AcquireError> {
+        self.ll_sem.acquire(1).await?;
+        Ok(SemaphorePermit {
             sem: &self,
             permits: 1,
-        }
+        })
     }
 
     /// Acquires `n` permits from the semaphore
```

Comment:
and maybe this should say something like
```suggestion
    /// Acquires `n` permits from the semaphore.
    /// 
    /// If the semaphore has been closed, this returns an [`AcquireError`].
    /// Otherwise, this returns a [`SemaphorePermit`] representing the 
    /// acquired permits.
```